diag_manager/diag_util.F90

```diff
@@ -634,8 +634,8 @@ SUBROUTINE log_diag_field_info(module_name, field_name, axes, long_name, units,&
     INTEGER, DIMENSION(:), INTENT(in) :: axes !< Axis IDs
     CHARACTER(len=*), OPTIONAL, INTENT(in) :: long_name !< Long name for field.
     CHARACTER(len=*), OPTIONAL, INTENT(in) :: units !< Unit of field.
-    REAL, OPTIONAL, INTENT(in) :: missing_value !< Missing value value.
-    REAL, DIMENSION(2), OPTIONAL, INTENT(IN) :: range !< Valid range of values for field.
+    CLASS(*), OPTIONAL, INTENT(in) :: missing_value !< Missing value value.
+    CLASS(*), DIMENSION(:), OPTIONAL, INTENT(IN) :: range !< Valid range of values for field.
```

Comment:
I guess we need to implement like this if we have data array passed in with dimension.

Reply:
Yes, I still don't understand the reason for this, but it allows the GNU unit tests to pass.